Fix/ruby style mitigations

nysthee opened the pull request:
@jnunemaker, I think this is ready. Commits after bbad00b are new fixes. Please review.

jnunemaker commented:
Left a few comments. Looks good other than those. How do you do this? What is the command you ran with rubocop or whatever? I dig consistency and might want to do this on other projects.

nysthee commented:
In the last commit I added the files I used for this particular project:
- .rubocop.yml: offences in this file while be ignored in rubocop's checks (I already added the things you didn't like)
- .rubocop_todo.yml: (generated with `rubocop  --auto-gen-config`) a kind of rubocop -report on the offences in the project

What I do is removing 1 offence a time from to-do and fix the given offence.  When the offence mentions `# Cop supports --auto-correct.` it means the offence can be dealt with automatically with `rubocop -a`.

Imo these files are great additions to any project. A rubopcop_todo.yml to show what things you want dealt with and rubocop.yml to enforce your style. Integrating a rubocop check in ci for example will make rubocop pass unless they made an offence against a the enforced style.

Being a big fan of the todo list I made a tiny feature for https://github.com/refactorcop/refactorcop (in pr) to add a badge to their site (http://refactorcop.com/, through which I found this project) showing wich project has a todo(showing the intention, the style is WIP) in their repo.

nysthee commented:
@jnunemaker if you want you can assign me to some other projects and I would be he happy to do the same thing.
